src/emulator/auth/operations.ts

```diff
@@ -2543,6 +2556,123 @@ function parsePendingCredential(
   return { user, signInProvider };
 }
 
+function createTenant(
+  state: ProjectState,
+  reqBody: Schemas["GoogleCloudIdentitytoolkitAdminV2Tenant"]
+): Schemas["GoogleCloudIdentitytoolkitAdminV2Tenant"] {
+  assert(state instanceof AgentProjectState, "((Can only create tenant in agent project.))");
+
+  const formattedPhoneNumbers: { [key: string]: string } = {};
+  if (reqBody.testPhoneNumbers) {
+    Object.entries(reqBody.testPhoneNumbers).forEach(([phoneNumber, value]) => {
+      formattedPhoneNumbers[formatPhoneNumber(phoneNumber)] = value;
+    });
+  }
+
+  const tenant = {
+    displayName: reqBody.displayName,
+    allowPasswordSignup: reqBody.allowPasswordSignup,
+    enableEmailLinkSignin: reqBody.enableEmailLinkSignin,
+    enableAnonymousUser: reqBody.enableAnonymousUser,
+    disableAuth: reqBody.disableAuth,
+    testPhoneNumbers: formattedPhoneNumbers,
+    mfaConfig: reqBody.mfaConfig,
+    tenantId: "", // Placeholder until one is generated
+  };
+
+  return state.createTenant(tenant);
+}
+
+function formatPhoneNumber(phoneNumber: string) {
+  assert(
+    phoneNumber.length < PHONE_NUMBER_MAX_LENGTH,
+    "TOO_LONG ((The string supplied was too long to parse.))"
+  );
+
+  // Only check that the phone number is reasonably formatted, but does not
+  // fully verify that phone number is valid.
+  // TODO: support unicode, alpha characters and extensions?
+  const validPhoneNumberPattern = new RegExp(
+    /^\d{2}$|^\+*(?:[ -x\.\[\]\(\)~\*]*\d){3,}[ -x\.\[\]\(\)~\*\d]*$/g
+  );
+  assert(validPhoneNumberPattern.test(phoneNumber), "INVALID_PHONE_NUMBER");
+
+  let normalizedNumber = phoneNumber.replace(/\D/g, "");
+  assert(
+    normalizedNumber.length <= MAX_LENGTH_FOR_NSN,
+    "TOO_LONG ((The string supplied is too long to be a phone number.))"
+  );
+  normalizedNumber = "+" + normalizedNumber;
+
+  return normalizedNumber;
+}
+
+function listTenants(
+  state: ProjectState,
+  reqBody: unknown,
+  ctx: ExegesisContext
+): Schemas["GoogleCloudIdentitytoolkitAdminV2ListTenantsResponse"] {
+  assert(state instanceof AgentProjectState, "((Can only list tenants in agent project.))");
+  const pageSize = Math.min(Math.floor(ctx.params.query.pageSize) || 20, 1000);
+  const tenants: Tenant[] = state.listTenants(ctx.params.query.pageToken);
```

Comment:
I believe the type of this variable can be inferred

Reply:
Ah yes yes, thanks! Removed the explicit typing